src/main/java/edu/harvard/iq/dataverse/dataaccess/SwiftAccessIO.java

```diff
@@ -48,43 +48,48 @@
 
     private static final Logger logger = Logger.getLogger("edu.harvard.iq.dataverse.dataaccess.SwiftAccessIO");
 
-    public SwiftAccessIO() {
-        this((T)null);
-    }
-
-    public SwiftAccessIO(T dvObject) {
-        this(dvObject, null);
-    }
-
-    public SwiftAccessIO(T dvObject, DataAccessRequest req) {
-        super(dvObject, req);
-
+	public SwiftAccessIO() {
+		//Partially functional StorageIO object - constructor only for testing
+		super();
+	}
+	
+    public SwiftAccessIO(T dvObject, DataAccessRequest req, String driverId) {
+        super(dvObject, req, driverId);
+        readSettings();
         this.setIsLocalFile(false);
     }
     
-    public SwiftAccessIO(String swiftLocation) {
-        this((T)null);
+    public SwiftAccessIO(String swiftLocation, String driverId) {
+    	super(swiftLocation, driverId);
+    	readSettings();
         this.swiftLocation = swiftLocation;
+        this.setIsLocalFile(false);
     }
 
-    private Account account = null;
+    private void readSettings() {
+    	isPublicContainer = Boolean.parseBoolean(System.getProperty("dataverse.files." + this.driverId + ".isPublicContainer", "true"));
+        swiftFolderPathSeparator = System.getProperty("dataverse.files." + this.driverId + ".folderPathSeparator", "_");
+        swiftDefaultEndpoint = System.getProperty("dataverse.files." + this.driverId + ".defaultEndpoint");
+        tempUrlExpires = Integer.parseInt(System.getProperty("dataverse.files." + this.driverId + ".temporaryUrlExpiryTime", "60"));
+		
+	}
+
+	private Account account = null;
     private StoredObject swiftFileObject = null;
     private Container swiftContainer = null;
-    private boolean isPublicContainer = Boolean.parseBoolean(System.getProperty("dataverse.files.swift.isPublicContainer", "true"));
-    private String swiftFolderPathSeparator = System.getProperty("dataverse.files.swift.folderPathSeparator", "_");
-    private String swiftDefaultEndpoint = System.getProperty("dataverse.files.swift.defaultEndpoint");
+    private boolean isPublicContainer = true;
+    private String swiftFolderPathSeparator = "_";
```

Comment:
This has also been hardcoded. @poikilotherm can probably answer if this is something needed by the community. I don't mind this being hard coded in general.